File: abc/abc323/a.cpp
// https://atcoder.jp/contests/abc323/submissions/46280102

#include <bits/stdc++.h>
using namespace std;

int main() {
  string s;
  cin >> s;

  for(int i = 1; i < s.size(); i += 2) {
    if(s[i] == '1') {
      cout << "No";
      return 0;
    }
  }

  cout << "Yes";
}

File: abc/abc323/a.java
// https://atcoder.jp/contests/abc323/submissions/47099132

import java.util.Scanner;

class Main {
  public static void main(String args[]) {
    Scanner stdIn = new Scanner(System.in);

    String s = stdIn.next();
    for(int i = 1; i < s.length(); i += 2) {
      if(s.charAt(i) != '0') {
        System.out.println("No");
        System.exit(0);
      }
    }

    System.out.println("Yes");
  }
}
